pkg/registry/core/shoot/storage/storage.go

```diff
@@ -134,6 +135,32 @@ func (r *StatusREST) Update(ctx context.Context, name string, objInfo rest.Updat
 	return r.store.Update(ctx, name, objInfo, createValidation, updateValidation, forceAllowCreate, options)
 }
 
+// BindingREST implements the REST endpoint for changing the binding of a Shoot.
+type BindingREST struct {
+	store *genericregistry.Store
+}
+
+var (
+	_ rest.Storage = &BindingREST{}
+	_ rest.Getter  = &BindingREST{}
+	_ rest.Updater = &BindingREST{}
+)
+
+// New creates a new (empty) internal Shoot object.
+func (r *BindingREST) New() runtime.Object {
+	return &core.Shoot{}
+}
+
+// Get retrieves the object from the storage. It is required to support Patch.
+func (r *BindingREST) Get(ctx context.Context, name string, options *metav1.GetOptions) (runtime.Object, error) {
```

Comment:
Thanks for adding this to enable `PATCH`. Should we now also add to the documentation, that migration can be triggered with the following request:
```
curl -k --cert ~/tmp/client.crt --key ~/tmp/client.key -XPATCH -H "Accept: application/json" -H "Content-Type: application/merge-patch+json" --data '{"spec":{"seedName":"<destination-seed-name>"}}' https://localhost:2443/apis/core.gardener.cloud/v1beta1/namespaces/<namespace>/shoots/<shoot-name>/binding
```

Here the authentication might be different - instead of certificates an authorization token could be used, depending on the current setup.